src/actions.rs

```diff
@@ -86,63 +144,18 @@ impl ActionHandler {
         match result {
             BuildResult::Success(ref x) | BuildResult::Failure(ref x) => {
                 debug!("build - Success");
+
+                self.clear_build_results();
+
                 {
                     let mut results = self.previous_build_results.lock().unwrap();
```

Comment:
Can you lock `previous_build_results` inside `parse_compiler_messages`, there seems no reason to hold the lock longer than that.